{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "dd85b163-6512-419e-91cb-c3a397a19782",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "Cannot use import statement outside a module",
     "execution_count": 11,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:1",
      "import OpenAI from 'openai';",
      "^^^^^^",
      "",
      "SyntaxError: Cannot use import statement outside a module",
      "    at new Script (node:vm:117:7)",
      "    at createScript (node:vm:269:10)",
      "    at Object.runInThisContext (node:vm:317:10)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "import OpenAI from 'openai';"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5471cc33-4237-44e9-82a4-9c5931c64cc5",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "Identifier 'OpenAI' has already been declared",
     "execution_count": 12,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:1",
      "const OpenAI = require('openai');",
      "^",
      "",
      "SyntaxError: Identifier 'OpenAI' has already been declared",
      "    at Script.runInThisContext (node:vm:137:12)",
      "    at Object.runInThisContext (node:vm:317:38)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "const OpenAI = require('openai');"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "df0f453b-8e6f-48fa-bed2-1e0579605ca3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{\n",
       "  ChatCompletionRequestMessageRoleEnum: [Getter],\n",
       "  ChatCompletionResponseMessageRoleEnum: [Getter],\n",
       "  CreateImageRequestSizeEnum: [Getter],\n",
       "  CreateImageRequestResponseFormatEnum: [Getter],\n",
       "  OpenAIApiAxiosParamCreator: [Getter],\n",
       "  OpenAIApiFp: [Getter],\n",
       "  OpenAIApiFactory: [Getter],\n",
       "  OpenAIApi: [Getter],\n",
       "  Configuration: [Getter]\n",
       "}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "OpenAI\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "b4b26234-ccb5-4812-aa08-5af8dbcb73ee",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "Identifier 'client' has already been declared",
     "execution_count": 14,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:1",
      "const client = new OpenAI({",
      "^",
      "",
      "SyntaxError: Identifier 'client' has already been declared",
      "    at Script.runInThisContext (node:vm:137:12)",
      "    at Object.runInThisContext (node:vm:317:38)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "const client = new OpenAI({\n",
    "  apiKey: process.env['OPENAI_API_KEY'], // This is the default and can be omitted\n",
    "});"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b30f916d-008a-476e-869d-19a6e61b1ec9",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "OpenAI is not a constructor",
     "execution_count": 15,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:1",
      "const client2 = new OpenAI({",
      "                ^",
      "",
      "TypeError: OpenAI is not a constructor",
      "    at evalmachine.<anonymous>:1:17",
      "    at Script.runInThisContext (node:vm:137:12)",
      "    at Object.runInThisContext (node:vm:317:38)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "const client2 = new OpenAI({\n",
    "  apiKey: process.env['OPENAI_API_KEY'], // This is the default and can be omitted\n",
    "});"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "82217573-491b-4423-a708-bb4092c7681d",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "Unexpected end of input",
     "execution_count": 16,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:7",
      "",
      "",
      "",
      "SyntaxError: Unexpected end of input",
      "    at new Script (node:vm:117:7)",
      "    at createScript (node:vm:269:10)",
      "    at Object.runInThisContext (node:vm:317:10)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    }
   ],
   "source": [
    "async function getCompletion() {\n",
    "  const response = await openai.completions.create({\n",
    "    model: 'text-davinci-003',\n",
    "    prompt: 'Hello, world!',\n",
    "    max_tokens: 50\n",
    "  });\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "97d674fd-b675-4d25-a045-75410e4eb735",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "Unexpected end of input",
     "execution_count": 17,
     "output_type": "error",
     "traceback": [
      "evalmachine.<anonymous>:6",
      "  });",
      "     ",
      "",
      "SyntaxError: Unexpected end of input",
      "    at new Script (node:vm:117:7)",
      "    at createScript (node:vm:269:10)",
      "    at Object.runInThisContext (node:vm:317:10)",
      "    at run ([eval]:1020:15)",
      "    at onRunRequest ([eval]:864:18)",
      "    at onMessage ([eval]:828:13)",
      "    at process.emit (node:events:518:28)",
      "    at emit (node:internal/child_process:950:14)",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    }
   ],
   "source": [
    "async function getCompletion() {\n",
    "  const response = await openai.completions.create({\n",
    "    model: 'text-davinci-003',\n",
    "    prompt: 'Hello, world!',\n",
    "    max_tokens: 50\n",
    "  });"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "87e4681d-3884-47cb-aec1-fcb2aa795cea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Promise {\n",
       "  <rejected> ReferenceError: openai is not defined\n",
       "      at getCompletion (evalmachine.<anonymous>:2:20)\n",
       "      at evalmachine.<anonymous>:11:1\n",
       "      at Script.runInThisContext (node:vm:137:12)\n",
       "      at Object.runInThisContext (node:vm:317:38)\n",
       "      at run ([eval]:1020:15)\n",
       "      at onRunRequest ([eval]:864:18)\n",
       "      at onMessage ([eval]:828:13)\n",
       "      at process.emit (node:events:518:28)\n",
       "      at emit (node:internal/child_process:950:14)\n",
       "      at process.processTicksAndRejections (node:internal/process/task_queues:83:21),\n",
       "  [Symbol(async_id_symbol)]: 2344,\n",
       "  [Symbol(trigger_async_id_symbol)]: 2341,\n",
       "  [Symbol(kResourceStore)]: undefined\n",
       "}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ReferenceError: openai is not defined\n",
      "    at getCompletion (evalmachine.<anonymous>:2:20)\n",
      "    at evalmachine.<anonymous>:11:1\n",
      "    at Script.runInThisContext (node:vm:137:12)\n",
      "    at Object.runInThisContext (node:vm:317:38)\n",
      "    at run ([eval]:1020:15)\n",
      "    at onRunRequest ([eval]:864:18)\n",
      "    at onMessage ([eval]:828:13)\n",
      "    at process.emit (node:events:518:28)\n",
      "    at emit (node:internal/child_process:950:14)\n",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "async function getCompletion() {\n",
    "  const response = await openai.completions.create({\n",
    "    model: 'text-davinci-003',\n",
    "    prompt: 'Hello, world!',\n",
    "    max_tokens: 50\n",
    "  });\n",
    "\n",
    "  console.log(response);\n",
    "}\n",
    "\n",
    "getCompletion();"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "188a2fba-1a63-4f00-97da-b647d572dc05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Promise {\n",
       "  <rejected> TypeError: Cannot read properties of undefined (reading 'create')\n",
       "      at getCompletion (evalmachine.<anonymous>:2:45)\n",
       "      at evalmachine.<anonymous>:11:1\n",
       "      at Script.runInThisContext (node:vm:137:12)\n",
       "      at Object.runInThisContext (node:vm:317:38)\n",
       "      at run ([eval]:1020:15)\n",
       "      at onRunRequest ([eval]:864:18)\n",
       "      at onMessage ([eval]:828:13)\n",
       "      at process.emit (node:events:518:28)\n",
       "      at emit (node:internal/child_process:950:14)\n",
       "      at process.processTicksAndRejections (node:internal/process/task_queues:83:21),\n",
       "  [Symbol(async_id_symbol)]: 2498,\n",
       "  [Symbol(trigger_async_id_symbol)]: 2495,\n",
       "  [Symbol(kResourceStore)]: undefined\n",
       "}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Cannot read properties of undefined (reading 'create')\n",
      "    at getCompletion (evalmachine.<anonymous>:2:45)\n",
      "    at evalmachine.<anonymous>:11:1\n",
      "    at Script.runInThisContext (node:vm:137:12)\n",
      "    at Object.runInThisContext (node:vm:317:38)\n",
      "    at run ([eval]:1020:15)\n",
      "    at onRunRequest ([eval]:864:18)\n",
      "    at onMessage ([eval]:828:13)\n",
      "    at process.emit (node:events:518:28)\n",
      "    at emit (node:internal/child_process:950:14)\n",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)"
     ]
    }
   ],
   "source": [
    "async function getCompletion() {\n",
    "  const response = await OpenAI.completions.create({\n",
    "    model: 'text-davinci-003',\n",
    "    prompt: 'Hello, world!',\n",
    "    max_tokens: 50\n",
    "  });\n",
    "\n",
    "  console.log(response);\n",
    "}\n",
    "\n",
    "getCompletion();"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "24b6e827-b862-4ba4-b76a-a54bb78efb7b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error: TypeError: Cannot read properties of undefined (reading 'create')\n",
      "    at getCompletion (evalmachine.<anonymous>:4:47)\n",
      "    at evalmachine.<anonymous>:16:1\n",
      "    at Script.runInThisContext (node:vm:137:12)\n",
      "    at Object.runInThisContext (node:vm:317:38)\n",
      "    at run ([eval]:1020:15)\n",
      "    at onRunRequest ([eval]:864:18)\n",
      "    at onMessage ([eval]:828:13)\n",
      "    at process.emit (node:events:518:28)\n",
      "    at emit (node:internal/child_process:950:14)\n",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Promise {\n",
       "  undefined,\n",
       "  [Symbol(async_id_symbol)]: 2795,\n",
       "  [Symbol(trigger_async_id_symbol)]: 2792,\n",
       "  [Symbol(kResourceStore)]: undefined\n",
       "}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "// Example usage to create a completion\n",
    "async function getCompletion() {\n",
    "  try {\n",
    "    const chatCompletion = await OpenAI.chat.completions.create({\n",
    "    messages: [{ role: 'user', content: 'Say this is a test' }],\n",
    "    model: 'gpt-4o',\n",
    "  });\n",
    "\n",
    "    console.log(chatCompletion);\n",
    "  } catch (error) {\n",
    "    console.error('Error:', error);\n",
    "  }\n",
    "}\n",
    "\n",
    "getCompletion();\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2f531403-cdaa-4008-9a6c-ead5b79a82bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error: TypeError: Cannot read properties of undefined (reading 'completions')\n",
      "    at getCompletion (evalmachine.<anonymous>:3:46)\n",
      "    at evalmachine.<anonymous>:14:1\n",
      "    at Script.runInThisContext (node:vm:137:12)\n",
      "    at Object.runInThisContext (node:vm:317:38)\n",
      "    at run ([eval]:1020:15)\n",
      "    at onRunRequest ([eval]:864:18)\n",
      "    at onMessage ([eval]:828:13)\n",
      "    at process.emit (node:events:518:28)\n",
      "    at emit (node:internal/child_process:950:14)\n",
      "    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Promise {\n",
       "  undefined,\n",
       "  [Symbol(async_id_symbol)]: 3261,\n",
       "  [Symbol(trigger_async_id_symbol)]: 3258,\n",
       "  [Symbol(kResourceStore)]: undefined\n",
       "}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)TypeError: Invalid URL\n",
      "    at new URL (node:internal/url:806:29)\n",
      "    at parseUrl (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:25:32)\n",
      "    at request (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:43:17)\n",
      "    at /app/node_modules/dd-trace/packages/dd-trace/src/telemetry/send-data.js:103:9\n",
      "    at ClientRequest.<anonymous> (/app/node_modules/dd-trace/packages/dd-trace/src/exporters/common/request.js:137:7)\n",
      "    at Object.onceWrapper (node:events:633:26)\n",
      "    at ClientRequest.emit (node:events:518:28)\n",
      "    at emitErrorEvent (node:_http_client:101:11)\n",
      "    at Socket.socketErrorListener (node:_http_client:504:5)\n",
      "    at Socket.emit (node:events:518:28)"
     ]
    }
   ],
   "source": [
    "async function getCompletion() {\n",
    "  try {\n",
    "    const chatCompletion = await OpenAI.chat.completions.create({\n",
    "      messages: [{ role: 'user', content: 'Say this is a test' }],\n",
    "      model: 'gpt-4',  // Correct model name\n",
    "    });\n",
    "\n",
    "    console.log(chatCompletion);\n",
    "  } catch (error) {\n",
    "    console.error('Error:', error);\n",
    "  }\n",
    "}\n",
    "\n",
    "getCompletion();\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d91ef71-6f4a-4d03-9b6d-eaf8a108c257",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "JavaScript (Node.js)",
   "language": "javascript",
   "name": "javascript"
  },
  "language_info": {
   "file_extension": ".js",
   "mimetype": "application/javascript",
   "name": "javascript",
   "version": "20.18.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
